src/tox_current_env/hooks.py

```diff
@@ -134,7 +135,11 @@ def tox_testenv_create(venv, action):
         target = sys.executable
         rm_venv(venv)
         os.makedirs(os.path.dirname(link))
-        os.symlink(target, link)
+        if sys.platform == "win32":
+            # Avoid requiring admin rights on Windows
+            subprocess.check_call('mklink /J "%s" "%s"' % (link, target), shell=True)
```

Comment:
Can you not use `shell=True`? Would the following work?
```suggestion
            subprocess.check_call(['mklink', '/J', link, target])
```
There's a [warning against shell=True](https://docs.python.org/3/library/subprocess.html#security-considerations)  in the subprocess docs, so I'd prefer not using it.

Reply:
> Can you not use shell=True? Would the following work?

Unfortunately, no. The `mklink` command is part of the CMD shell, like e.g. `dir`, `cd` etc. Those cannot be called unless you specify `shell=True`.

Reply:
I see.
And can you use `shlex.quote`, as mentioned in the subprocess docs?

Reply:
As a matter of style, I'd prefer an f-string (or even a format method call) over `%s`, to much the surrounding code base.

Reply:
> I see.
> And can you use `shlex.quote`, as mentioned in the subprocess docs?

Yes, but I need to convert the single quotation marks into double quotation marks:

```python
link = "C:\Users\fredrik\code\repos\mylittleci\.tox\formatting\Scripts\python"
target = "c:\users\fredrik\code\repos\mylittleci\venv\scripts\python.exe"

# Unfortunately, shlex.quote wraps the string in single quotes ('), which needs to be changed to double quotes (")
# Link is converted to type str due to the fact that it comes in as type LocalPath
quoted_link = shlex.quote(str(link)).replace("'", "\"")
quoted_target = shlex.quote(target).replace("'", "\"")

cmd = f'mklink /J {quoted_link} {quoted_target}'
subprocess.check_call(cmd, shell=True)
```

```python
>>> print(cmd)
mklink /J "C:\Users\fredrik\code\repos\mylittleci\.tox\formatting\Scripts\python" "c:\users\fredrik\code\repos\mylittleci\venv\scripts\python.exe"
```

I would trust the facilities which provide the `link` and `target` values in this case, and not bother with `shlex.quote`:

```python
cmd = f'mklink /J {link} {target}'
```

or perhaps

```python
from pathlib import Path
cmd = f'mklink /J {Path(link)} {Path(target)}'
```

...but I am not sure if pathlib brings anything to the table in this case.

Reply:
> Yes, but I need to convert the single quotation marks into double quotation marks...

Why is that needed?

> I would trust the facilities which provide the link and target values in this case...

Can Windows paths contain `"`?

Reply:
Yes, Windows paths can be wrapped in `"` to support e.g. spaces:

```
cd "C:\Program Files"
```

Reply:
Correction: Powershell supports single quotes. But Python's subprocess relies on CMD, not Powershell.

Reply:
My concern was: What if the path you are in contains a `"` character in the directory name. Is that possible? If yes, we need to use some escaping.

Reply:
> My concern was: What if the path you are in contains a `"` character in the directory name. Is that possible? If yes, we need to use some escaping.

![image](https://user-images.githubusercontent.com/994357/89628932-4d813580-d89d-11ea-8ea9-beff6a5fd98e.png)

Reply:
In that case, I suppose enclosing the filenames in `"` should do, WDYT @encukou?

Reply:
Well, it sounds OK to me, though I don't know much about Windows. (The system refuses to create a symlink for security reasons, but then does allow you to create it by running command? I'm baffled.)

I won't really be able to fix any issues in this code.

Reply:
I'm honestly buffled as well :D 

We already claim that this might not work on Windows, so maybe it's fine if it doesn't work for another reason. As long as we keep saying that, I think that merging this doesn't do any harm. (I don't want to drive contributors away just because we don't know Windows.)

If this breaks and @fredrikaverpil isn't around to fix it, we can always remove it.

Reply:
Yeah, I find this weird as well. The docs mention that there are workarounds in Python 3.8 but they entail setting the entire Windows installation to this "Developer mode" setting ... so yeah, no - not much of a workaround.

I haven't looked at the source code but I wonder if it is Microsoft which have somehow decided symlinking is dangerous...